Lib/asyncio/base_events.py

```diff
@@ -248,6 +248,8 @@ def __init__(self):
         self.slow_callback_duration = 0.1
         self._current_handle = None
         self._task_factory = None
+        self._tasks = weakref.WeakSet()
```

Comment:
I think there should be an `unregister_task()` method instead of relying on GC and WeakSet. It's weird if holding on to a copy of `all_tasks()` prevents any of those tasks from leaving the set even if they're finished. (The WeakSet implementation was fine for an internal-only debugging tool, but if it's going to be public I think it should have more well-defined semantics.)

Reply:
I think it's a good idea.  `asyncio.Task` can have a `__del__` method which would call `self.loop.unregister_task(self)`.

Reply:
Unfortunately it doesn't work: if tasks are stored in strong `set()` inside a loop the `__del__` is never called -- refcount never became zero.
I could call `_unregister_task()` from `_step()` on exit from coroutine.

Note about naming: `current_task()` and `all_tasks()` are public methods but `_register_task()` and family should be never called by user code, that's why I've added underscore prefix.